src/main/java/Duke.java

```diff
@@ -1,10 +1,235 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
 public class Duke {
+
+    private final String horizontalLine = "-------------------------";
+    private ArrayList<Task> taskList;
+    private Scanner sc;
+    private static File saveFile = new File("duke.txt");
+    
+    public Duke() {
+        this.taskList = new ArrayList<>();
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        duke.greeting();
+        duke.run();
+        duke.goodBye();
+    }
+    //Solution below adapted from https://github.com/24Donovan24/ip
+    
+    private void run() {
+        if (saveFile.exists()) {
+            load();
+        } else {
+            initialise();
+        }
+    }
+
+    public void load() {
+        try {
+            sc = new Scanner(this.saveFile);
+            while (sc.hasNextLine()) {
+                String taskInFile = sc.nextLine();
+                String[] taskInArray = taskInFile.split(" \\| "); //The \\ is really equivalent to a single \ (the first \ is required as a Java escape sequence in string literals).
+                // It is then a special character in regular expressions which means "use the next character literally, don't interpret its special meaning"
+                String taskType = taskInArray[0];
+                switch (taskType) {
+                    case "T": {
+                        Task task = new ToDo(taskInArray[2]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "D": {
+                        Task task = new Deadline(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "E": {
+                        Task task = new Event(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
```

Comment:
Since in each case you are marking the task as done when loading them, maybe u can abstract the process out?